include/aws/auth/credentials.h

```diff
@@ -758,13 +776,9 @@ struct aws_credentials *aws_credentials_new(
  * @return a valid credentials object, or NULL
  */
 AWS_AUTH_API
-struct aws_credentials *aws_credentials_new_with_account_id(
```

Comment:
I added the function recently, https://github.com/awslabs/aws-c-auth/pull/260 therefore it's safe to remove.